### PR TITLE
add loading spinner for lookdev

### PR DESCRIPTION
**Pull request by JT00y**
## References
closes #_insert number here_

### CODE REVIEW

**Comment by HexaField** [comment]
is this still necessary after this PR? https://github.com/ir-engine/ir-engine/pull/455

**Comment by JT00y** [comment]
> is this still necessary after this PR? #455

yes, this is for the lookdev asset which not use model component, it just to overwrite the value of certian components, therefore I make reactor to listen snapshot.

**Comment by HexaField** [comment]
> > is this still necessary after this PR? #455
> 
> yes, this is for the lookdev asset which not use model component, it just to overwrite the value of certian components, therefore I make reactor to listen snapshot.

Ok. After #455 is in, we should move this logic there, rather than adding even more complexity to the loading logic. This will help keep functionality together and avoid creating more and more nested reactors.

File: packages/editor/src/functions/addMediaNode.ts [hunk 1]
```diff
@@ -169,10 +194,28 @@ export async function addMediaNode(
...
+          const fileNameWithExt = pathArray[lastIndex]
+          const fileNameArray = fileNameWithExt.split('.')
+          const name = decodeURI(fileNameArray[0])
+          setComponent(entity, NameComponent, name)
+
+          reactor.stop()
+        }, [modelComponent])
+
+        return null
+      })
```
Comment (dinomut1) [inline packages/editor/src/functions/addMediaNode.ts]: let's remove this from this PR, not in scope

**Review by dinomut1 — CHANGES_REQUESTED** [review]

**Comment by dinomut1** [comment]
> > > is this still necessary after this PR? #455
> > 
> > 
> > yes, this is for the lookdev asset which not use model component, it just to overwrite the value of certian components, therefore I make reactor to listen snapshot.
> 
> Ok. After #455 is in, we should move this logic there, rather than adding even more complexity to the loading logic. This will help keep functionality together and avoid creating more and more nested reactors.

That won't work as the lookdev objects don't use the model component, or any other component that can be queried by a system. Let's leave this as is

File: packages/editor/src/functions/addMediaNode.ts (hunk 1 above)
Comment (JT00y) [reply packages/editor/src/functions/addMediaNode.ts:217]: okay i will do this, there is a fix for load lookdev asset with multiple lookdev components, i will leave tthis in this pr

File: packages/editor/src/functions/addMediaNode.ts [hunk 2]
```diff
@@ -126,25 +121,54 @@ export async function addMediaNode(
...
+        useEffect(() => {
+          if (loadingEntity.value !== UndefinedEntity && snapshots.length !== length) {
+            removeEntityNodeRecursively(loadingEntity.value)
+            loadingEntity.set(UndefinedEntity)
+            if (reactor) reactor.stop()
+          }
+        }, [snapshots.length])
+        return null
+      })
+      return null
```
Comment (dinomut1) [inline packages/editor/src/functions/addMediaNode.ts]: I don't think we need this reactor. We should just keep this simple, add a loading spinner before the gltfloader.load function, and then remove it in the onLoad callback, and add an onError callback that removes it as well.

**Review by dinomut1 — CHANGES_REQUESTED** [review]

**Review by dinomut1 — APPROVED** [review]

**Pull request merged**